### PR TITLE
HPCC-21846 Remote file timeouts producing misleading error message

### PR DESCRIPTION
**Pull request by shamser**
Signed-off-by: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by shamser** [comment]
Testing:
1) Shutdown dafilesrv service
2) Submit a request to WsWorkunits/WUResult ESP service with the following parameters:
   <LogicalName>rttemp::testkeyedjoin</LogicalName>
   <Cluster>mythor</Cluster>
   <SuppressXmlSchema>1</SuppressXmlSchema>
   <BypassCachedResult>1</BypassCachedResult>


The bug may generate an error message like:
1) <Message>2019-04-04 09:57:21 GMT: CreateIFile cannot resolve //**0.0.0.0**/var/lib/HPCCSystems/hpcc-data/thor/rttemp/testkeyedjoin._3_of_4</Message>

_Note: Node IP have been corrupted so that it is using 0.0.0.0_

OR
2) ESP will fail to return any messages and will show seg faults in the logs

Subsequent requests will fail even if dafilesrv is restarted.

With the fix, a message like this will be returned:
<Message>2019-04-04 09:57:21 GMT: CreateIFile cannot resolve //**172.17.0.2**/var/lib/HPCCSystems/hpcc-data/thor/rttemp/testkeyedjoin._3_of_4</Message>

_Note: Node IP has not been corrupted__

Once dafilesrv has been restarted and suitable amount of time has elapsed for the socket to be marked available again, the request will suceeded.

**Comment by shamser** [comment]
@jakesmith Please can you review

**Comment by mckellyln** [comment]
@shamser just a comment -
>> suitable amount of time has elapsed for the socket to be marked available again
I recall some discussion on this specific part before.  I think after a failed connection to dafilesrv that host is marked 'down' and for the next 15 minutes a client will not attempt any new connections to it - and perhaps that is too long.  Either a background thread checking connectivity (ie like the soapcall blacklisting code) or just a shorter time to wait might be good to consider.

**Comment by shamser** [comment]
@mckellyln I see that.  I haven't dug too much into this so far.  Can you tell me if this blacklisting period is configurable?

**Comment by mckellyln** [comment]
I think the SocketDeblacklister() is a pooled thread that tries 10 times to connect() which each connect() timeout set to 10 sec with a delay between each failed connect() that grows by 5 sec.  From a quick glance, I don't see where those time values can be configured at runtime.
The dafilesrv connect() after host marked down wait period is also, I think, a macro define of 15 minutes.

**Comment by richardkchapman** [comment]
typo (timeouts)

**Comment by shamser** [comment]
@mckellyln I'll look into making the blacklisting period configurable in a separate ticket.   Please can you review these changes.

File: common/fileview2/fvdisksource.cpp [hunk 1]
```diff
@@ -131,27 +131,25 @@ bool PhysicalFileInfo::readData(MemoryBuffer & out, __int64 startOffset, size32_
...
-            cachedIO.setown(cachedFile->open(IFOread));
-        }
-        catch (IException * e)
+        for (unsigned copy=0; copy<2; copy++)
         {
-            e->Release();
+            try
+            {
+                RemoteFilename rfn;
+                cachedFile.setown(createIFile(dfp->getFilename(rfn, copy)));
```
Comment (mckellyln) [inline common/fileview2/fvdisksource.cpp]: Isn't this getFilename(copy, rfn) ?

**Comment by mckellyln** [comment]
@shamser I am fine with these changes.  I don't really see any functional change however.

**Review by mckellyln — APPROVED** [review]
@shamser I am fine with these changes. I don't really see any functional change however.

**Comment by shamser** [comment]
@richardkchapman Mark's reviewed the changes. Please can you merge changes.

**Comment by richardkchapman** [comment]
@shamser It looks like a reasonable change from a code cleanliness point of view but I'm not sure I see how it fixes the issue referenced in the linked Jira?

**Comment by shamser** [comment]
@richardkchapman I am not sure now if the crashes are related to this fix.  I am trying to reproduce the orignal crashes.  I'll get back to you.

**Comment by richardkchapman** [comment]
@shamser Assigning to you for now

**Comment by shamser** [comment]
@richardkchapman The crashes were unrelated to this issue. I've re-edited subject in jira and commit.

**Pull request merged**